tests/test_frame.py

```diff
@@ -3556,3 +3558,9 @@ class MyDict(TypedDict):
     my_dict = MyDict(a="", b="")
     sr = pd.Series(my_dict)
     check(assert_type(sr, pd.Series), pd.Series)
+
+
+def test_series_empty_dtype() -> None:
+    """Test for the creation of a Series from an empty list GH571."""
+    new_tab: Sequence[Never] = []
```

Comment:
For the test, it should work without setting a type for `new_tab`.  So if you change this line to `new_tab = []`, does the empty list match `Sequence[Never]` ?

also, double check that passing an empty string will NOT match this, because strings are sequences, and I'm not sure if `""` would be an OK match for `Sequence[Never]`

Reply:
Thanks for the feedback, in order:
- `mypy` raises an error when instantiating an empty sequence so I needed to annotate it
```
===========================================
Beginning: 'Run mypy on 'tests' (using the local stubs) and on the local stubs'
===========================================

tests/test_frame.py:3566: error: Need type annotation for "new_tab" (hint: "new_tab: list[<type>] = ...")  [var-annotated]
Found 1 error in 1 file (checked 226 source files)
```
- I have added a test for the empty string

Reply:
FYI this is the error that you get when trying to typehint `""` as a `Sequence[Never]` so the types are well separated.
```
tests/test_frame.py:3571: error: Incompatible types in assignment (expression has type "str", variable has type "Sequence[Never]")  [assignment]
```
for mypy above and pyright below
```
Diagnostics: 
Type "Literal['']" is not assignable to declared type "Sequence[Never]"  
"Literal['']" is not assignable to "Sequence[Never]"                         
Type parameter "_T_co@Sequence" is covariant, but "str" is not a subtype of "Never"
Type "str" is not assignable to type "Never" [reportAssignmentType]
```

Reply:
OK, I see that you added an explicit test for `pd.Series([])`, so the issue with `new_tab` is nothing to worry about.